omc3/scripts/kmod_summary.py

```diff
@@ -76,144 +99,235 @@ def output_kmod_summary_tables(
+
+    Returns:
+        list[str]: List of formatted string tables. Includes separate sections for X-plane, Y-plane, and optionally averaged results.
+    """
+    LOG.info("Creating .txt tables.")
+
+    kmod_summary_x = kmod_summary[[IP_COLUMN, NAME_COLUMN] + COLS_X]
+    kmod_summary_y = kmod_summary[[IP_COLUMN, NAME_COLUMN] + COLS_Y]
+    kmod_summary_x_averaged = kmod_summary_averaged[[IP_COLUMN] + COLS_X]
+    kmod_summary_y_averaged = kmod_summary_averaged[[IP_COLUMN] + COLS_Y]
+
+    table_logbook = []
+    if kmod_summary_lumiimb:
+        table_logbook.append(f"============================================ Luminosity Imbalance ====================================================\n{kmod_summary_lumiimb}\n")
+    for plane, df in [("X", kmod_summary_x), ("Y", kmod_summary_y)]:
+        table_str = df.to_string()
+        table_logbook.append(f"\n============================================ {BEAM_DIR}{beam} Results ({plane}-plane) ====================================================\n\n{table_str}\n")
+    if kmod_summary_x_averaged is not None and kmod_summary_y_averaged is not None:
+        for plane, df_averaged in [("X", kmod_summary_x_averaged), ("Y", kmod_summary_y_averaged)]:
+            table_str_averaged = df_averaged.to_string()
+            table_logbook.append(f"\n======================================== {BEAM_DIR}{beam} Averaged Results ({plane}-plane) ===============================================\n\n{table_str_averaged}\n")
+    return table_logbook
 
 def _save_outputs(
-    beam: str,
+    beam: int,
     table_logbook: list[str],
     summary_df: tfs.TfsDataFrame,
     save_output_dir: Path | str,
     ) -> None:
 
     """
     Save logbook text output and .tfs summary for a given beam.
+
     Args:
-        beam (str): beam, ex. B1, B2
+        beam (int): Beam number to process.
         table_logbook (list[str]): Single .txt file.
         summary_df (tfs.TfsDataFrame): Single .tfs dataframe.
         save_output_dir (Path | str ): Path to the saving output directory.
     """
 
-    logbook_table_path = save_output_dir / f"{beam}_tables_logbook.txt"
-    summary_path = save_output_dir / f"{beam}_kmod_summary.tfs"
+    logbook_table_path = save_output_dir / f"{BEAM_DIR}{beam}_kmod_summary.txt"
+    summary_path = save_output_dir / f"{BEAM_DIR}{beam}_kmod_summary{EXT}"
     with logbook_table_path.open("w") as f:
         f.write("\n".join(table_logbook))
     tfs.write(summary_path, summary_df)
-    LOG.info(f"Writing logbook summary output file {logbook_table_path}.")
-    LOG.info(f"Writing .tfs summary output file {summary_path}.")
+    LOG.info(f"Writing .txt summary output file {logbook_table_path}.")
+    LOG.info(f"Writing {EXT} summary output file {summary_path}.")
     return
 
 def _logbook_entry(
+    beam: int,
     logbook: str,
-    beam: str,
-    save_output_dir: Path | str,
+    logbook_file: str | list[str]
```

Comment:
As I understand from reading below, this is **not** a file. Let's rename.
It might also make sense for the caller to take care of providing the text directly themselves (and do the `"\n".join(...)` themselves)